### PR TITLE
Source package generation script

### PR DESCRIPTION
**Pull request by tomaszrybicki**

### CODE REVIEW

File: package.cmake [hunk 1]
```diff
@@ -18,13 +18,28 @@ endif()
...
 
+# Source package
+set(CPACK_SOURCE_GENERATOR TGZ)
+# Exclude untracked and ignored files, including those in submodules
+execute_process(
+    COMMAND bash -c "git clean --dry-run -xd -e VERSION | cut -d ' ' -f 3- | xargs -I{} printf {}\; \
+    && git submodule foreach --recursive --quiet 'git ls-files --ignored --directory --others --exclude-standard \
+    | sed \"s|^|$path/|\"' | xargs -I{} printf {}\; \
+    && git submodule foreach --recursive --quiet 'git ls-files --others --directory --exclude-standard \
+    | sed \"s|^|$path/|\"' | xargs -I{} printf {}\;"
```
Comment (tomaszrybicki) [inline package.cmake]: Sorry

File: package.cmake [hunk 2]
```diff
@@ -18,13 +18,27 @@ endif()
...
+execute_process(
+    COMMAND bash -c "git ls-files --directory --others --exclude-standard -x VERSION | xargs -I{} printf \`pwd\`/{}\; \
+    && git ls-files --directory --ignored --others --exclude-standard | xargs -I{} printf {}\; \
+    && git submodule foreach --recursive --quiet 'git ls-files --ignored --directory --others --exclude-standard \
+    | xargs -I{} printf \`pwd\`/{}\;'\
+    && git submodule foreach --recursive --quiet 'git ls-files --others --directory --exclude-standard \
+    | xargs -I{} printf \`pwd\`/{}\;'"
+    OUTPUT_VARIABLE CPACK_SOURCE_IGNORE_FILES
+    WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+    ERROR_QUIET
```
Comment (tomaszrybicki) [inline package.cmake]: Sorry

File: version.cmake [hunk 3]
```diff
@@ -4,12 +4,23 @@ set(IOTRACE_VERSION_MINOR 0)
...
+    COMMAND bash -c "git log --pretty=format:%h -n 1 | tr -d '\n'"
     OUTPUT_VARIABLE IOTRACE_VERSION_LABEL
     WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
     ERROR_QUIET
 )
 
+if (NOT IOTRACE_VERSION_LABEL)
+    message(STATUS "git repository not found, trying label from VERSION file")
+    execute_process(
+        COMMAND bash -c "tail -n 1 ${CMAKE_CURRENT_LIST_DIR}/VERSION | awk -F '=' '{print $2}' \
```
Comment (mariuszbarczak) [inline version.cmake]: Can we grep by LABEL instead of taking "tail -n 1"

File: package.cmake [hunk 4]
```diff
@@ -18,13 +18,27 @@ endif()
 execute_process(OUTPUT_VARIABLE uname_r OUTPUT_STRIP_TRAILING_WHITESPACE COMMAND uname -r)
 
 
+# Source package
+set(CPACK_SOURCE_GENERATOR TGZ)
+# Exclude untracked and ignored files, including those in submodules
+execute_process(
```
Comment (mariuszbarczak) [inline package.cmake]: do we pack third party cmake?

**Pull request merged**